compiler/passes/flattenFunctions.cpp

```diff
@@ -93,6 +93,11 @@ passableByVal(Type* type) {
 // Otherwise passing by value is more efficient.
 static bool
 passByRef(Symbol* sym) {
+
+  // Is it a non-POD record type?
+  if( sym->type->symbol->hasFlag(FLAG_NOT_POD) )
+    return true;
+
```

Comment:
I am OK with the above addition going in. However, it seems like it is not the right property to test for.
Please add a comment to the code justifying the relationship of POD-ness to passByRef().